service/insights_v3.go

```diff
@@ -0,0 +1,24 @@
+package service
+
+import (
+	. "github.com/talbright/go-desk/types"
+	. "github.com/talbright/go-desk/resource"
+	"net/http"
+)
+
+type InsightsV3Service struct {
+	client *Client
+}
+
+func (s *InsightsV3Service) Report(query *InsightsV3Query) (*InsightsV3Report, *http.Response, error) {
```

Comment:
To get a better idea of how this is working in the API I wrote a quick script to exercise it: 

https://gist.github.com/talbright/48ec6fa8522b7fa4ce32

So basically yes the insightsv3 endpoint will set the response code to 429. 

Secondly, even though the insightsv3 API endpoint isn't setting the ```X-Rate-Limit-Reset```, it's still being sent back (base controller or filtering is doing that in the server side).

So in the end, your work in #13 will trigger and delay for ```X-Rate-Limit-Reset``` seconds and retry. Even though thats not directly related to the concurrency limits of insightsv3 endpoints, it still provides good error handling with retries.

:+1: